core/src/storage/impls/owned_node_set.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Conflux Foundation. All rights reserved.
+// Conflux is free software and distributed under GNU General Public License.
+// See http://www.gnu.org/licenses/
+
+/// A container to store information about owned nodes.
+#[derive(Default, Debug)]
+pub struct OwnedNodeSet {
+    dirty: BTreeMap<ActualSlabIndex, Option<DeltaMptDbKey>>,
+    committed: BTreeSet<DeltaMptDbKey>,
```

Comment:
Owned trie node is always dirty, before committing. We never "own" a committed trie node, unless the db commit fails in the middle then you have to revert the transaction and clean up the node in memory.

Reply:
Can you confirm? cow_node_ref.rs has code `owned_node_set.insert(committed_node_ref.clone(), None);`, so a committed node can be in the owned node set.

Reply:
Yes I can confirm. You can see that the statement you mention is exactly in the committing phase. These nodes were held in OwnedNodeSet only in case these nodes should be freed up later.

The extra code you wrote probably works fine but there is no need to over generalize.